### PR TITLE
Extract persistence related code to a new shared crate

### LINKED ISSUE #1492: Extract database feature from common crate
## Why
We want to reduce the dependencies needed by our projects, reduce the number of features of `mithril-common` and make our codebase more manageable by splitting code in cohesive blocks.

## What
Extract all store & database related code from `mithril-common` to a new local crate . This crate doesn't needs to be published to crates.io since the mithril-client doesn't use those features.

## How

- [x] Create a new local crate (name to be defined, ie: `mithril-database` or `mithril-store` ?)
- [x] Add it as dependency to the nodes that needs it (should be only `mithril-aggregator` and `mithril-signer`)
- [x] Move the database & store related code from `mithril-common` to this new crate
- [x] Update the imports in impacted project to target the new crate
- [x] Remove the `database` feature from `mithril-common`

### PR DESCRIPTION
**Pull request by Alenar**
## Content

This PR create a new shared crate, `mithril-persistence`. All all store & database related code from `mithril-common` has been moved to it.

This makes enhance our code base maintainability, reduce the dependencies of the mithril-common crate, allow to remove its `database` feature.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [x] Update README file (if relevant)
  - [x] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)
Closes #1492.

### CODE REVIEW

File: mithril-persistence/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Mithril-persistence !
+
+**This is a work in progress** 🛠 
+
+This crate contains shared api and helpers for Mithril nodes that needs to persist data.
```
Comment (jpraynaud) [inline mithril-persistence/README.md:5]:
```suggestion
This crate contains shared api and helpers for Mithril nodes which need to persist data.
```

File: mithril-persistence/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Mithril-persistence !
```
Comment (jpraynaud) [inline mithril-persistence/README.md:1]:
```suggestion
# Mithril-persistence
```

File: mithril-persistence/src/database/signed_entity_hydrator.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+use mithril_common::entities::{Beacon, Epoch, SignedEntityType, SignedEntityTypeDiscriminants};
+
+use crate::sqlite::HydrationError;
+
+/// Helper struct to hydrate [SignedEntityType].
+pub struct SignedEntityTypeHydrator {}
+
+impl SignedEntityTypeHydrator {
+    /// Create an [SignedEntityType] from data coming from the database
```
Comment (jpraynaud) [inline mithril-persistence/src/database/signed_entity_hydrator.rs:11]:
```suggestion
    /// Create a [SignedEntityType] from data coming from the database
```

File: mithril-persistence/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#![warn(missing_docs)]
+
+//! Shared datatypes and traits used to for the persistence layer of Mithril rust projects
```
Comment (jpraynaud) [inline mithril-persistence/src/lib.rs:3]:
```suggestion
//! Shared datatypes and traits used to for the persistence layer of Mithril nodes
```

**Review by jpraynaud — APPROVED** [review]
LGTM :+1: 

I left few comments for typos below. Also, should we bump the minor version of `mithril-common` as we changed the available features in the crate?

**Comment by Alenar** [comment]
> Also, should we bump the minor version of `mithril-common` as we changed the available features in the crate?

Yup, bumping the minor version of `mithril-common` was on my mind .

**Review by dlachaumepalo — APPROVED** [review]

**Pull request merged**